src/Storages/MergeTree/MergeTask.cpp

```diff
@@ -251,12 +252,23 @@ bool MergeTask::ExecuteAndFinalizeHorizontalPart::prepare()
         }
     }
 
-    global_ctx->new_data_part->setColumns(global_ctx->storage_columns, infos, global_ctx->metadata_snapshot->getMetadataVersion());
-
     const auto & local_part_min_ttl = global_ctx->new_data_part->ttl_infos.part_min_ttl;
     if (local_part_min_ttl && local_part_min_ttl <= global_ctx->time_of_merge)
         ctx->need_remove_expired_values = true;
 
+    if (global_ctx->data->getSettings()->allow_experimental_block_number_column
+        && !ctx->need_remove_expired_values && global_ctx->metadata_snapshot->getProjections().empty()
```

Comment:
Why `!ctx->need_remove_expired_values`? It's relate to TTL and it will just filter our some rows after merge. Should be ok with block number columns.

Reply:
ya you are right, fixed it. Updated to not persist `_block_number` column if `groupByTTL` exists